### PR TITLE
Fix: aligin config create to be managed by apps with Dispatch function

### PR DESCRIPTION
**Pull request by wonderflow**
Signed-off-by: Jianbo Sun <jianbo.sjb@alibaba-inc.com>

### Description of your changes

This PR migrate the config provider package from workflow to kubevela.

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/controller/core.oam.dev/v1alpha2/application/generator.go [hunk 1]
```diff
@@ -95,6 +97,9 @@ func (h *AppHandler) GenerateApplicationSteps(ctx monitorContext.Context,
 			Apply:  h.Dispatch,
 			Delete: h.Delete,
 		})
+	configprovider.Install(handlerProviders, h.r.Client, func(ctx context.Context, resources []*unstructured.Unstructured, applyOptions []apply.ApplyOption) error {
```
Comment (barnettZQG) [inline pkg/controller/core.oam.dev/v1alpha2/application/generator.go:100]: Move to after `generator.GenerateRunners`, and make sure this provider can not be replaced.

Comment (wonderflow) [reply pkg/controller/core.oam.dev/v1alpha2/application/generator.go:100]: No, it must register first. The registered job won't be replaced when exists.

**Review by barnettZQG — APPROVED** [review]
LGTM

File: pkg/utils/apply/apply.go [hunk 2]
```diff
@@ -121,16 +122,16 @@ func loggingApply(msg string, desired client.Object, quiet bool) {
...
+func trimLastAppliedConfigurationForSpecialResources(desired client.Object) bool {
 	if desired == nil {
 		return false
 	}
 	gvk := desired.GetObjectKind().GroupVersionKind()
 	gp, kd := gvk.Group, gvk.Kind
 	if gp == "" {
 		// group is empty means it's Kubernetes core API, we won't record annotation for Secret and Configmap
-		if kd == "Secret" || kd == "ConfigMap" || kd == "CustomResourceDefinition" {
+		if kd == "Secret" || kd == "ConfigMap" || kd == "CustomResourceDefinition" || kd == "Grafana" {
```
Comment (Somefive) [inline pkg/utils/apply/apply.go:134]: I think we do not to modify it here

**Review by Somefive — APPROVED** [review]

**Pull request merged**